frontend/hooks/hypercerts-client.ts

```diff
@@ -2,17 +2,52 @@ import {
   DEFAULT_CHAIN_ID,
   NFT_STORAGE_TOKEN,
   WEB3_STORAGE_TOKEN,
+  OVERRIDE_CHAIN_NAME,
+  OVERRIDE_GRAPH_NAME,
+  OVERRIDE_GRAPH_BASE_URL,
+  OVERRIDE_GRAPH_NAMESPACE,
+  CONTRACT_ADDRESS,
+  UNSAFE_FORCE_OVERRIDE_CONFIG,
 } from "../lib/config";
-import { HypercertClient } from "@hypercerts-org/sdk";
+import { HypercertClient, HypercertClientConfig } from "@hypercerts-org/sdk";
 import { ethers } from "ethers";
 import { useEffect, useState } from "react";
 import { useAccount, Connector } from "wagmi";
 
-const defaultClient = new HypercertClient({
+const clientConfig: Partial<HypercertClientConfig> = {
   chainId: DEFAULT_CHAIN_ID,
   nftStorageToken: NFT_STORAGE_TOKEN,
   web3StorageToken: WEB3_STORAGE_TOKEN,
-});
+};
+
+function loadOverridingConfig(clientConfig: Partial<HypercertClientConfig>) {
```

Comment:
Not sure if it's the scope for this PR, but the env var config is becoming a little messy. Should we put more responsibility in the SDK or the FE?

We now have:
- SDK looking at ENV vars in case there are no overrides
- next.config.js loading env vars from NEXT_APP_BLA_BLA to BLA_BLA
- OVERRIDE vars in config.ts
- loadOverridingConfig in this file. 

Cognitively, it feels like to many places for env vars and overrides

Reply:
Yeah. I didn't want to handle env var clean up in this PR, but I did need to be able to override most, if not all, things.

Reply:
@bitbeckers in thinking about this, I do believe the SDK should _receive_ configuration from the consumer as opposed to expecting them over environment. It'd be much more explicit that way when used by an outside developer. The frontend should be the one doing the mapping of the environment variables. I think this makes it a cleaner for us as well so we handle environment variables in less places. It might be a little more _verbose_ between the sdk and the frontend, but at least it's very explicit.